lib/cartopy/mpl/geoaxes.py

```diff
@@ -1276,8 +1276,10 @@ def imshow(self, img, *args, **kwargs):
 
         return result
 
-    def gridlines(self, crs=None, draw_labels=False, xlocs=None,
-                  ylocs=None, dms=False, **kwargs):
+    def gridlines(self, crs=None, draw_labels=False,
+                  xlocs=None, ylocs=None, dms=False,
+                  x_inline=None, y_inline=None,
```

Comment:
Maybe a keyword to choose the kind of labels would be appropriate : "inline", "outline", "both".

Reply:
I think in most cases, "outline" would be preferred and "both" would be very cluttered in my opinion. auto_inline defaults to True which will decide if inline or outline (probably mostly outline), but if users want to override it, then they can just set x_inline=True and y_inline=True.